src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1791,7 +2109,11 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
                 }
 
                 Debug.Assert(!IsConditionalState);
-                ResultType = InferResultNullability(binary, leftType, rightType);
+                SetResultType(binary, InferResultNullability(binary, leftType, rightType));
+
+                // For nested binary operators, this can be the only time they're visited due to explicit stack used in AbstractFlowPass.VisitBinaryOperator,
+                // so we need to set the flow-analyzed type here.
+                SetAnalyzedNullability(binary, _visitResult);
```

Comment:
>SetAnalyzedNullability(binary, _visitResult); [](start = 16, length = 45)

Is `_visitResult` different than the return value of `InferResultNullability()` above? If not, this call can be removed and replaced with `updateAnalyzedNullability: true` above.